app/subscribers/publish_zync_event_subscriber.rb

```diff
@@ -10,25 +10,36 @@ def initialize(publisher = DEFAULT_PUBLISHER)
 
   attr_reader :publisher
 
-  # @param [ZyncEvent] event
-  def call(event)
-    unless ThreeScale.config.onpremises
+  delegate :domain_event_in_saas?, :build_zync_event, to: 'self.class'
+
+  class << self
+    def domain_event_in_saas?(event)
+      ThreeScale.saas? && (event.is_a?(Domains::ProxyDomainsChangedEvent) || event.is_a?(Domains::ProviderDomainsChangedEvent))
```

Comment:
We basically skip domain changes in SaaS becase we don't need to create routes there, correct? How about hosted apicast on-prem though? Any point to consider it as well?

Reply:
> We basically skip domain changes in SaaS becase we don't need to create routes there, correct?

Exactly.

> How about hosted apicast on-prem though? Any point to consider it as well?

Mmm, not sure I understand the concern. In on-premises the `Domains::*` events will be processed and ZyncEvent will be created for them, thus triggering a ZyncWorker, and Zync processing the event (so, if needed, the routes will be created).